### PR TITLE
Deprecate the default rooms beta for #admins and #announce rooms

### PR DESCRIPTION
**Pull request by jasperhuangg**
### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/280874

#### Before performing any local tests
- Delete ` || canUseAllBetas(betas)` from the condition in [`canUseDefaultRooms`](https://github.com/Expensify/App/blob/75d8eff70b24c605aed6bfefe40899aec527ee0a/src/libs/Permissions.js#L38-L44).
- Create an account in OldDot. Go to Settings > Domains and create a new domain.
- Sign into that same account in NewDot. Create a workspace, or use an existing one. Be sure to have the #announce and #admins rooms ready.
- Make sure you're using the Web Proxy

#### LHN Options
1. Verify that the #admins room for your workspace appears in the LHN
2. Verify that the #announce room for your workspace appears in the LHN.
3. Verify that the #domain.com room does not appear in the LHN
4. Add back the ` || canUseAllBetas(betas)` to the condition in [`canUseDefaultRooms`](https://github.com/Expensify/App/blob/75d8eff70b24c605aed6bfefe40899aec527ee0a/src/libs/Permissions.js#L38-L44).
5. Verify that the #domain.com room appears in the LHN.

#### Last viewed report
If you added ` || canUseAllBetas(betas)` to the condition in [`canUseDefaultRooms`](https://github.com/Expensify/App/blob/75d8eff70b24c605aed6bfefe40899aec527ee0a/src/libs/Permissions.js#L38-L44), delete it again.

1. Navigate to the #admins room.
2. Sign out and back into your account.
3. Verify you are taken to the #admins room after signing in.
4. Navigate to the #admins room.
5. Sign out and back into your account.
6. Verify you are taken to the #announce room after signing in.

### Offline tests
N/A

### QA Steps

ping @jasperhuangg to QA

#### LHN Options
1 Verify that the #admins room for your workspace appears in the LHN
2. Verify that the #announce room for your workspace appears in the LHN.
3. Verify that the #domain.com room does not appear in the LHN

#### Last viewed report
1. Navigate to the #admins room.
2. Sign out and back into your account.
3. Verify you are taken to the #admins room after signing in.
4. Navigate to the #admins room.
5. Sign out and back into your account.
6. Verify you are taken to the #announce room after signing in.

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/31285285/236312398-cbbb86d9-57ca-41b4-85d9-329e67acc1d4.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/31285285/236313933-fd5acc68-4597-479d-b2af-f79ea30d5a04.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/31285285/236314740-f35a7983-5834-41e5-a6c4-1c2a6b8f34b1.mov

</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-05-04 at 12 50 49 PM

</details>

<details>
<summary>iOS</summary>

Screenshot 2023-05-04 at 12 53 20 PM

</details>

<details>
<summary>Android</summary>

Screenshot 2023-05-04 at 12 53 36 PM

</details>

### CODE REVIEW

**Comment by jasperhuangg** [comment]
Asked about test domains [here](https://expensify.slack.com/archives/C9YU7BX5M/p1683174413360089) (see QA steps for context)

File: src/libs/OptionsListUtils.js [hunk 1]
```diff
@@ -843,7 +843,6 @@ function getMemberInviteOptions(
     return getOptions([], personalDetails, {
         betas,
         searchInputValue: searchValue.trim(),
-        excludeDefaultRooms: true,
```
Comment (jasperhuangg) [inline src/libs/OptionsListUtils.js:846]: Just removed this because I found that this param wasn't being used anywhere in `getOptions`

**Comment by jasperhuangg** [comment]
Adding CP Staging label as there's some urgency around this

**Review by yuwenmemon — APPROVED** [review]
LGTM! But why CP to staging? Can this not just follow the normal deploy process?

**Comment by jasperhuangg** [comment]
@yuwenmemon Yeah it can, just thought there was some urgency around this. If you don't agree feel free to remove the label.

**Comment by yuwenmemon** [comment]
@jasperhuangg yeah better to err on the side of caution IMO. If someone wants it _right now_ we can always CP it later too.

**Comment by jasperhuangg** [comment]
Ah @0xmiroslav you need to be an internal employee to do that on your dev environment. You can just test the changes to admins and announce rooms for now. Ignore anything that has anything to do with domain rooms.

**Comment by PauloGasparSv** [comment]
Hey @jasperhuangg I think the steps under "Last viewed report" repeat, is that's intentional?

**Comment by jasperhuangg** [comment]
> Hey @jasperhuangg I think the steps under "Last viewed report" repeat, is that's intentional?

@PauloGasparSv fixed them, was a typo, thanks for pointing that out!

**Comment by jasperhuangg** [comment]
@0xmiroslav will you be able to compete your review today?

**Review by 0xmiroslav — APPROVED** [review]
LGTM 🎉 
#admins and #announce rooms work well. not able to test domain rooms.

**Pull request merged**